scripts/codegen/cogapp/cogapp.py

```diff
@@ -0,0 +1,781 @@
+# coding: utf8
```

Comment:
This is present on PyPi: https://pypi.org/project/cogapp/. Do we really need to include it here or could we add it to `requirements.txt`?

Reply:
what do you mean @b0661? Why do you need to import it from PyPi and make a local copy of it? if you put `cogapp` in `requirements.txt` the module will be installed automatically when the user sets up the development environment and the Zephyr scripts will always have access to the latest version. I don't understand why we wouldn't do the same thing we do with all other Python modules we require, which is to install them via PyPi. Having local copies is always a maintenance burden.

Reply:
The local copy is reduced to the bare minimum and  only very litlle is used anymore. The rest will be replaced over time. There is no extra burden as there is no benefit for codegen to follow cogapp. It is vice versa; it would be an extra burden to follow cogapp.

Reply:
OK, that was not what I understood originally when you said "cogapp.py is the original import from PyPi. It was not changed...". Fine then, if we want to avoid following upstream to do our own thing I'm OK with that.